BitmovinAnalyticsCollector/Classes/Collector/stateMachines/StateMachine.swift

```diff
@@ -43,7 +43,7 @@ public class StateMachine {
         print("Generated Bitmovin Analytics impression ID: " +  impressionId.lowercased())
     }
 
-    public func transitionState(destinationState: PlayerState, time: CMTime?) {
+    public func transitionState(destinationState: PlayerState, time: CMTime?, data: [AnyHashable : Any]? = nil) {
         if state == destinationState {
             return
         } else if state == .buffering && destinationState == .qualitychange {
```

Comment:
Can't we instead of that `if` and `if else` checks have simply one guard check for the `state == .buffering && destinationState == .qualitychange`. This is the only thing that matters for us here, as far as I see.

Reply:
yeah that would be enough in this case, as we don't do anything but return. But in our web collector there are a lot more different conditions and also more logic, so it's probably better we keep it like this in case there's more to come in future